scipy/special/tests/test_basic.py

```diff
@@ -1445,6 +1445,20 @@ def test_betainc_domain_errors(self, func, args):
             with pytest.raises(special.SpecialFunctionError, match='domain'):
                 special.betainc(*args)
 
+    @pytest.mark.parametrize('dtype', [np.float32, np.float64])
+    def test_gh21426(self, dtype):
+        # Test for gh-21426: betaincinv must not return NaN
+        a = np.array([5.], dtype=dtype)
+        x = np.array([0.5], dtype=dtype)
+        result = special.betaincinv(a, a, x)
+        if sys.maxsize <= 2**32 and dtype == np.float32:
+            # result is not very precise for 32bit architecture and single
```

Comment:
The significand only has around 7 digits of precision for single precision floats, so this isn’t unusual. One thing we could do is express the tolerance as a multiple of `np.finfo(dtype).eps`

Reply:
Ah, good point. I think I only used double precision so far in scipy stuff. Will try out `10*np.finfo(dtype).eps` later today.